### ISSUE TITLE
`Annotated` generic appears to erase callability

### ISSUE THREAD

**Issue by davesque** [microsoft/pyright#7126]
I recently noticed that the `HttpUrl` type imported from the popular pydantic library is reported as not being callable by pyright.  Here's a minimal example that mimics the code in pydantic and produces the error:

```python
# example.py
from typing import Annotated, Self

class Url:
    def __new__(cls, url: str) -> Self:
        ...

HttpUrl = Annotated[Url, "foo"]

url = HttpUrl("https://example.com")
```

Reported error:
```
example.py:10:7 - error: Object of type "type[Url]" is not callable (reportCallIssue)
```

Pyright playground link:
https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoCCKKYMAhjAKYAmANFAMqUA2wAUGwMbNkDOvUAKohmALjZRJUapWBQA%2BvJSUA7ooAU3XvQCuI0VF4wQASigBaAHyMWwcVIdQAdC44AJGDATDmUALyExKQUNADaPvQARMBgYJEAuhx6vgEeXj7qkQAWngi8ogD0BZQAHmTYzJROnJCRJkA

Is this correct or should the callability of the `Url` class be transmitted through the `Annotated` generic?

**Comment by erictraut** [comment]
There's an active discussion about this [in the typing forum](https://discuss.python.org/t/is-annotated-compatible-with-type-t/43898/14). 

The problem is that pydantic is using `Annotated` here in a way that I don't think was intended. `Annotated` is a special form that was meant to be used in type annotations, but in your code sample you're using it in a value expression — and attempting to call it. It appears that the runtime implementation of `Annotated` does call through to the `__call__` method of its first type argument, but that behavior is highly questionable because the first type argument could be something like `int | str` which is not callable. Also, `Annotated` doesn't appear to call through to any other magic methods, so it's not usable in other value expressions like an argument to `isinstance`.

I'm waiting for the discussion to reach a conclusion, but the consensus so far appears to be in favor of making these uses of `Annotated` illegal. Regardless, we should update the typing spec so it is clear on the intended behavior.

**Comment by davesque** [comment]
Interesting.  And it goes with a trend I've seen a couple times now where the pydantic project seems to be steering the type checking conversation in python ecosystem pretty forcefully.

**Comment by davesque** [comment]
So then I guess the `_BaseGenericAlias.__call__` implementation that gets resolved in the mro is not really intended to act as a pass-through to the `__init__` method of the origin class?

**Comment by erictraut** [comment]
The discussion in the typing forum converged on an answer that was adopted by the Typing Council. Pyright's current behavior matches the behavior in this clarification and the update typing spec.
